### PR TITLE
Overhaul cmake interface to ES2K SDE

### PR DESCRIPTION
**Pull request by ffoulkes**
- Replace the es2k-driver.cmake include file with FindEs2kDriver.cmake, a find-module that allows us to use `find_package(Es2kDriver)` to define namespaced cmake targets for the ES2K SDE.

- Define an `add_es2k_target_libraries()` function that provides an abstract interface to set the properties a target needs in order to compile and link with the ES2K SDE.

- Update krnlmon submodule reference.

Depends on https://github.com/ipdk-io/krnlmon/pull/49.

### CODE REVIEW

**Comment by ffoulkes** [comment]
_Nota bene:_ The CI build failure is a consequence of the original PR having been separated into three commits.

CI does a *DPDK* build on an *ES2K* commit. The build fails because the (unmodified) DPDK driver package does not define the target names the (modified) version of krnlmon expects.

File: cmake/FindEs2kDriver.cmake [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# FindEs2kDriver.cmake - import ES2K P4 Driver (SDE).
+#
+# Copyright 2022-2023 Intel Corporation
```
Comment (5abeel) [inline cmake/FindEs2kDriver.cmake:3]: New file shouldnt have the 2022 copyright(?)

File: cmake/FindEs2kDriver.cmake [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#-----------------------------------------------------------------------
+# Define SDE_INCLUDE_DIR
+#-----------------------------------------------------------------------
+find_path(SDE_INCLUDE_DIR
+    NAMES "bf_switchd/bf_switchd.h"
+    PATHS ${SDE_INSTALL_DIR}/include
+)
+if(NOT SDE_INCLUDE_DIR)
+  message(FATAL_ERROR "bf_switchd.h not found")
```
Comment (5abeel) [inline cmake/FindEs2kDriver.cmake:18]: Why are we looking specifically for the bf_switchd.h? Or is it just one file that was picked as that was common across all targets?

File: cmake/FindEs2kDriver.cmake (hunk 1 above)
Comment (ffoulkes) [reply cmake/FindEs2kDriver.cmake:3]: I suppose I could consider this to be an original file. It's a complete rewrite of `es2k-driver.cmake`, with very few similarities (and those so obvious that one could hardly claim to have created them).

File: cmake/FindEs2kDriver.cmake (hunk 2 above)
Comment (ffoulkes) [reply cmake/FindEs2kDriver.cmake:18]: Just looking for something to verify that SDE_INSTALL_DIR points to the root of an SDE install tree and that seemed unlikely to change.

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**